webui/react/src/components/ExperimentRetainLogsModal.tsx

```diff
@@ -0,0 +1,142 @@
+  onSubmit?: (successfulIds?: number[]) => void;
+}
+
+const ExperimentRetainLogsModalComponent: React.FC<Props> = ({
+  excludedExperimentIds,
+  experimentIds,
+  filters,
+  onSubmit,
+}: Props) => {
+  const idPrefix = useId();
+  const { openToast } = useToast();
+  // const [disabled, setDisabled] = useState<boolean>(true);
+  const [checked, setChecked] = useState<boolean>(false);
+  const [form] = Form.useForm<FormInputs>();
+
+  const handleCheckBoxChange = (event: CheckboxChangeEvent) => {
+    const isChecked = event.target.checked;
+    setChecked(isChecked);
+    if (isChecked) {
+      form.setFieldValue('numDays', -1);
+    }
+  };
+
+  const handleSubmit = async () => {
+    const values = await form.validateFields();
+    const numberDays = values.numDays;
+    if (excludedExperimentIds?.size) {
+      filters = { ...filters, excludedExperimentIds: Array.from(excludedExperimentIds) };
+    }
+
+    const results = await changeExperimentLogRetention({
+      experimentIds,
+      filters,
+      numDays: numberDays,
+    });
+
+    onSubmit?.(results.successful);
+
+    const numSuccesses = results.successful.length;
+    const numFailures = results.failed.length;
+
+    const stringDays = numberDays > 0 ? `for ${numberDays} days` : 'forever';
+
+    if (numFailures === 0) {
+      openToast({
+        closeable: true,
+        description: `Retained logs for ${results.successful.length} ${pluralizer(
+          experimentIds.length,
+          'experiment',
+        )} ${stringDays}`,
+        title: 'Retain Logs Success',
+      });
+    } else if (numSuccesses === 0) {
+      openToast({
+        description: `Unable to retain logs for ${numFailures} ${pluralizer(
+          experimentIds.length,
+          'experiment',
+        )}`,
+        severity: 'Warning',
+        title: 'Retain Logs Failure',
+      });
+    } else {
+      openToast({
+        closeable: true,
+        description: `Failed to retain logs for ${numFailures} experiments out of ${
+          numFailures + numSuccesses
+        } for ${numberDays} days.`,
+        severity: 'Warning',
+        title: 'Partial Retain Logs Failure',
+      });
+    }
+  };
+
+  return (
+    <Modal
+      cancel
+      size="small"
+      submit={{
+        //disabled,
+        form: idPrefix + FORM_ID,
+        handleError,
+        handler: handleSubmit,
+        text:
+          filters !== undefined
+            ? 'Retain Logs'
+            : `Retain logs for ${pluralizer(experimentIds.length, 'Experiment')}`,
+      }}
+      title={
+        filters !== undefined
+          ? 'Retain Logs'
+          : `Retain Logs for ${pluralizer(experimentIds.length, 'Experiment')}`
+      }>
+      <Form form={form} id={idPrefix + FORM_ID} layout="vertical">
+        <Form.Item
+          label="Number of Days"
+          name="numDays"
+          rules={[
+            {
+              message: 'Number of days is required',
+              min: checked ? -1 : 0,
```

Comment:
prob declare a constant for `-1` to make it more meaningful

Reply:
done!